### PR TITLE
Add openh264 library for darwin and windows

### PR DESCRIPTION
**Pull request by at-wat**
Use dockercore/golang-cross to cross compile openh264.
Set `-tags dynamic` to use system installed library.

**~Windows/Darwin binary is not tested yet.~**

### Reference issue
Partially address #82 
Non-x86_64 arch is still missing.

### CODE REVIEW

**Comment by at-wat** [comment]
Added a workaround to build with both MinGW MSYS 5.x and 6.x.
I confirmed that openh264 codec works on Windows 10.

**Comment by at-wat** [comment]
Hope someone could test darwin version on MacOS.

**Comment by lherman-cs** [comment]
Hey, I just tested it on mac, and I got this:

```
ld: warning: ignoring file ../../pkg/codec/openh264/../../../cvendor/lib/openh264/libopenh264.x86_64-darwin.a, file was built for archive which is not the architecture being linked (x86_64): ../../pkg/codec/openh264/../../../cvendor/lib/openh264/libopenh264.x86_64-darwin.a
Undefined symbols for architecture x86_64:
  "_WelsCreateSVCEncoder", referenced from:
      _enc_new in _x004.o
  "_WelsDestroySVCEncoder", referenced from:
      _enc_free in _x004.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

It looks like that the problem is caused by the linker conflict since mac uses XCode by default instead of GNU. I'm double checking it with GNU toolchain instead.

Also, I found that the struct initialization that I used in `bridge.cpp` is not compatible with Clang. So, I made a small fix here,  https://github.com/pion/mediadevices/pull/86.

**Comment by lherman-cs** [comment]
It doesn't seem that GNU `ld` supports Darwin sadly:
* https://www.quora.com/How-do-I-get-GNU-ld-on-OS-X
* https://github.com/Homebrew/homebrew-core/issues/17794
* https://apple.stackexchange.com/questions/341105/ld-linker-is-missing-when-installing-gcc-using-homebrew-on-mac-os

File: Makefile [hunk 1]
```diff
@@ -3,16 +3,53 @@ src_dir = $(vendor_dir)/src
...
 
-vendor: create_out_dirs openh264
+make_args.x86_64-windows = \
+	CC=x86_64-w64-mingw32-gcc \
+	CXX=x86_64-w64-mingw32-g++ \
+	ARCH=x86_64 \
+	OS=mingw_nt
+make_args.x86_64-darwin = \
+	CC=o64-clang \
+	CXX=o64-clang++ \
```
Comment (at-wat) [inline Makefile]: maybe `AR=llvm-ar` is required

Comment (at-wat) [reply Makefile]: This is it. Cross compiling the example is fixed by this.

**Comment by at-wat** [comment]
@lherman-cs I think the updated .a file should work for Darwin.

File: Makefile (hunk 1 above)
Comment (lherman-cs) [reply Makefile]: Yep, it works!

**Comment by lherman-cs** [comment]
Btw, I just found another small bug... Somehow, the errno in `enc_new` was non-zero when it returned. I created another fix here, https://github.com/pion/mediadevices/pull/87.

File: cvendor/builder.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM dockercore/golang-cross
```
Comment (lherman-cs) [inline cvendor/builder.Dockerfile]: I didn't know that this existed... TIL

File: Makefile [hunk 3]
```diff
@@ -3,16 +3,54 @@ src_dir = $(vendor_dir)/src
...
+	docker build -t mediadevices-builder -f cvendor/builder.Dockerfile .
+	docker run -i --rm \
+		-v $(CURDIR):/go/src/github.com/pion/mediadevices \
+		mediadevices-builder make $(lib_dir)/openh264/libopenh264.x86_64-linux.a
+	docker run -i --rm \
+		-v $(CURDIR):/go/src/github.com/pion/mediadevices \
+		mediadevices-builder make $(lib_dir)/openh264/libopenh264.x86_64-windows.a
+	docker run -i --rm \
+		-v $(CURDIR):/go/src/github.com/pion/mediadevices \
+		mediadevices-builder make $(lib_dir)/openh264/libopenh264.x86_64-darwin.a
```
Comment (lherman-cs) [inline Makefile]: `-i` doesn't seem necessary.

**Comment by at-wat** [comment]
Removed unnecessary docker build option and squashed archive fix into the first commit.

**Pull request merged**